pkg/kubelet/kubeletconfig/controller.go

```diff
@@ -108,9 +109,20 @@ func NewController(defaultConfig *kubeletconfig.KubeletConfiguration,
 
 // Bootstrap attempts to return a valid KubeletConfiguration based on the configuration of the Controller,
 // or returns an error if no valid configuration could be produced. Bootstrap should be called synchronously before StartSync.
-func (cc *Controller) Bootstrap() (*kubeletconfig.KubeletConfiguration, error) {
+func (cc *Controller) Bootstrap() (config *kubeletconfig.KubeletConfiguration, reterr error) {
 	utillog.Infof("starting controller")
 
+	// set metrics when all is said and done
+	var using string
```

Comment:
`using` is a bad name, is says nothing and spans a huge scope

Also, if you change the logic below later, it may be used uninitialized

Reply:
I don't understand that global defer at all, why not leave setting metrics local?